models/users.js

```diff
@@ -138,5 +139,15 @@ module.exports = {
     // FROM:  [ { user2: 123}, { user2: 456}, ... ]
     // TO:    [ 123, 456 ]
     return res.rows.map(row => row.user2)
+  },
+
+  // Autheticates user by checking secret and userId. Returns true if both values are returned by query.
+  authenticateUser: async function (userId, secret) {
+    const auth = {
+      text: 'SELECT id, secret FROM users WHERE id = $1 AND secret = $2',
+      values: [userId, secret]
+    }
+    const res = await db.query(auth)
+    return (res.rows.length !== 0)
```

Comment:
[nit] there's no need to SELECT secret in addition to id since it's not being returned
[nit] I feel like authenticateUser should be a service level function rather than model-level, because it has to do with interpreting data, but what you have here is a lot more straightforward which is fine too. Just a thought.